lib/cylc/task_events_mgr.py

```diff
@@ -869,3 +877,78 @@ def _setup_custom_event_handlers(self, itask, event, message):
                         cmd,
                     ),
                     retry_delays))
+
+    def _reset_job_timers(self, itask):
+        """Set up poll timer and timeout for task."""
+        if itask.state.status not in TASK_STATUSES_ACTIVE:
+            # Reset, task not active
+            itask.timeout = None
+            itask.poll_timer = None
+            return
+        ctx = (itask.submit_num, itask.state.status)
+        if itask.poll_timer and itask.poll_timer.ctx == ctx:
+            return
+        # Set poll timer
+        # Set timeout
+        timeref = None  # reference time, submitted or started time
+        timeout = None  # timeout in setting
+        delays = []  # polling intervals
+        if itask.state.status == TASK_STATUS_RUNNING:
+            timeref = itask.summary['started_time']
+            timeout_key = 'execution timeout'
+            timeout = self._get_events_conf(itask, timeout_key)
+            delays = self.get_host_conf(
+                itask, 'execution polling intervals', skey='job',
+                default=[900])  # Default 15 minute intervals
+            if itask.summary[self.KEY_EXECUTE_TIME_LIMIT]:
+                time_limit = itask.summary[self.KEY_EXECUTE_TIME_LIMIT]
+                try:
+                    host_conf = self.get_host_conf(itask, 'batch systems')
+                    batch_sys_conf = host_conf[itask.summary['batch_sys_name']]
+                except (TypeError, KeyError):
+                    batch_sys_conf = {}
+                time_limit_delays = batch_sys_conf.get(
+                    'execution time limit polling intervals', [60, 120, 420])
+                timeout = time_limit + sum(time_limit_delays)
+                # Remove execessive polling before time limit
+                while sum(delays) > time_limit:
+                    del delays[-1]
+                # But fill up the gap before time limit
+                if delays:
+                    size = int((time_limit - sum(delays)) / delays[-1])
+                    delays.extend([delays[-1]] * size)
+                time_limit_delays[0] += time_limit - sum(delays)
+                delays += time_limit_delays
+        else:  # if itask.state.status == TASK_STATUS_SUBMITTED:
```

Comment:
The comment here is deliberate. It is saying that the `else:` statement:

```
else:  # if itask.state.status == TASK_STATUS_SUBMITTED:
```

is equivalent to:

```
elif itask.state.status == TASK_STATUS_SUBMITTED:
```

The bare `else:` is faster because it does not have to evaluate `itask.state.status == TASK_STATUS_SUBMITTED`.

Reply:
I understood this, my point is that it wasn't immediately obvious to me that this comment was implying equivalence (hence the suggestion 'i.e.') as opposed to it being something left there during dev. Though it could easily just be me it was not at first evident to! Sorry if I was not clear.

Reply:
I see your point @sadielbartholomew but I think we use this idiom fairly often (possibly omitting the "if" in the comment though)

Reply:
Apologies, it is clearly confusion arising from my lack of experience - when I see similar comments I will now know distinctly what they mean.

Reply:
No apologies necessary! (and I wouldn't too surprised if you see a few things we do that aren't clear or correct).